tests/integration/test_branch_protection.py

```diff
@@ -0,0 +1,162 @@
+def fixture_branch_for_commit_not_signed_fail(
+    forked_github_branch: Branch, forked_github_repository: Repository
+):
+    """Create branch for the test_commit_not_signed_fail test.
+
+    Makes the branch the default branch for the repository and makes the latest commit unsigned.
+    """
+    # Make an unsigned commit
+    forked_github_repository.create_file(
+        "test.txt", "testing", "some content", branch=forked_github_branch.name
+    )
+
+    # Change default branch so that the commit is ignored by the check for unique commits being
+    # signed
+    forked_github_repository.edit(default_branch=forked_github_branch.name)
+
+    branch_with_protection = BranchWithProtection(
+        require_code_owner_reviews=False,
+        dismiss_stale_reviews_enabled=False,
+        bypass_pull_request_allowance_disabled=True,
+    )
+
+    test_branch_protection.edit(
+        branch=forked_github_branch, branch_with_protection=branch_with_protection
+    )
+
+    yield forked_github_branch
+
+    forked_github_branch.remove_protection()
+
+
+@pytest.mark.parametrize(
+    "forked_github_branch", [f"test-branch/branch/unsigned-commit/{uuid4()}"], indirect=True
+)
+def test_commit_not_signed_fail(
+    branch_for_commit_not_signed_fail: Branch,
+    forked_github_repository: Repository,
+):
+    """
+    arrange: given a branch that is compliant including a signed commit only in CI (on local runs
+        the branch has no unique commits and hence the check for unsigned commits will pass).
+    act: when branch_protection is called with the name of the branch.
+    assert: then a fail report is returned.
+    """
+    # Can't use branch_for_commit_not_signed_fail branch since it doesn't know about the latest
+    # commit
+    commit_sha = forked_github_repository.get_commit(
+        sha=branch_for_commit_not_signed_fail.name
+    ).sha
+
+    # The github_client is injected
+    report = branch_protection(  # pylint: disable=no-value-for-parameter
+        repository_name=forked_github_repository.full_name,
+        branch_name=branch_for_commit_not_signed_fail.name,
+        commit_sha=commit_sha,
+    )
+
+    assert report.result == Result.FAIL
+    assert report.reason, "expected a reason along with the fail result"
+    assert_.substrings_in_string(
+        ("commit", "not signed", commit_sha, branch_for_commit_not_signed_fail.name), report.reason
+    )
+
+
+@pytest.mark.parametrize(
+    "github_branch", [f"test-branch/branch/protected/{uuid4()}"], indirect=["github_branch"]
+)
+def test_pass(protected_github_branch_with_commit_in_ci: Branch, github_repository_name: str):
+    """
+    arrange: given a branch that is compliant including a signed commit only in CI (on local runs
+        the branch has no unique commits and hence the check for unsigned commits will pass).
+    act: when branch_protection is called with the name of the branch.
+    assert: then a pass report is returned.
+    """
```

Comment:
The arrange and act of this test is exactly the same as the previous one, but the assert is different.
The 2 tests are using slightly different fixtures in this one we don't change default branch.
I think something should be adapted in the arrange

Reply:
that is a mistake, the previous test actually has the latest commit as not signed. Should be fixed on next push